content/python/concepts/scope/scope.md

```diff
@@ -0,0 +1,58 @@
+---
+Title: 'Scope'
+Description: 'The variable is only usable inside the region it is created is called as scope in python'
+Subjects:
+  - 'Computer Science'
+  - 'Data Science'
+  - 'Web Development'
+Tags:
+  - 'Lists'
+  - 'Arrays'
+  - 'Objects'
+  - 'Data Structures'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+The variable is only usable inside the region it is created is called as **scope** in python. Now this region can be anything a class, method(function), loops, if statements, etc.
+
+## Types of Scopes
+ **Local Scope**: 
+ 
+ Suppose there is a function and a variable is initialized in it. Now this variable can only be used within that funtion not from outside the function.
+
+ ~~~
+def fun():
+    x = 200
+    print(x)
+
+myfun()
+ ~~~
+
+**Nexted Functions Scope**:
+
+Here, if the variable initialised in the outer function and that variable is used in the inner function then its totally fine, but when the case is vice versa it will generate an exception.
```

Comment:
```suggestion
In the example below, a variable `x` is defined within the local scope of the `outerfun()` function, followed by a defined `innerfun()` function. Since `innerfun()` exists within the local scope of `outerfun()`, `x` can be accessed and printed within `innerfun()`:

```

We always want to surround a code snippet with an empty line on the top and the bottom.